### DISCUSSION TITLE
Vine support unocss👀❓

### DISCUSSION THREAD

**Discussion by CodeGetters** [vue-vine/vue-vine#44] [Ideas]
I see vine is similar to react which compiles the returned string into html, if so is it compatible with unocss?

Can you give me an example if it is already unocss compatible? Because it didn't work when I used it

```vine
// app.vine
  return vine`
    <button class="random-pick-post-btn bg-red" @click="randomPick">
      Random pick a post
    </button>
    <BlogPost :id="id" />
  `
```

```ts
// uno.config.ts
// uno.config.ts
import { defineConfig, presetUno } from 'unocss'

export default defineConfig({
  presets: [
    presetUno(),
  ],
})
```

**Comment by ShenQingchuan** [comment]
Good first discussion！This is related to our ecosystem support and I'll put it into backlog~

**Reply by CodeGetters** [reply]
I guess since vine returns a template string, I guess the vine compiler can parse the class name and the vue project in the returned string🤔

**Reply by ShenQingchuan** [reply]
Hi @CodeGetters 

I'm so happy to tell you that UnoCSS integration is already done! You can check the example usage in `packages/playground`

But UnoCSS VSCode extension doesn't not work for now, the reason may be that UnoCSS doesn't know vine tagged string is a template HTML. I'll put it into backlog and solve it later.

You must use it with configuration! Because UnoCSS doesn't scan `.js`/`.ts` files **as default**. Check the docs description [here](https://unocss.dev/guide/extracting#extracting-from-build-tools-pipeline)

**Reply by CodeGetters** [reply]
That's cool, but the project didn't work after I pulled back the warehouse.
```shell
pnpm install
pnpm run dev
# or
pnpm run ext:dev

# there is an error
pnm run play
```

error message
```test
failed to load config from E:\software_cache\codetotal\vue\repo\vue-vine\packages\playground\vite.config.ts
error when starting dev server:
Error: Cannot find module 'E:\software_cache\codetotal\vue\repo\vue-vine\packages\vue-vine\node_modules\@vue-vine\vite-plugin\dist\index.js'
    at createEsmNotFoundErr (node:internal/modules/cjs/loader:1096:15)
    at finalizeEsmResolution (node:internal/modules/cjs/loader:1089:15)
    at resolveExports (node:internal/modules/cjs/loader:565:14)
    at Module._findPath (node:internal/modules/cjs/loader:634:31)
    at Module._resolveFilename (node:internal/modules/cjs/loader:1061:27)
    at Module._load (node:internal/modules/cjs/loader:920:27)
    at Module.require (node:internal/modules/cjs/loader:1141:19)
    at require (node:internal/modules/cjs/helpers:110:18)
    at Object.<anonymous> (E:\software_cache\codetotal\vue\repo\vue-vine\packages\vue-vine\dist\vite\index.js:3:18)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
E:\software_cache\codetotal\vue\repo\vue-vine\packages\playground:
 ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  @vue-vine/playground@ dev: `vite --port 3333 --open`
Exit status 1
 ELIFECYCLE  Command failed with exit code 1.
```

**Reply by ShenQingchuan** [reply]
Seems like you didn't get the bundle output? Check if there's a `dist` folder inside `packages/vite-plugin` and `packages/vue-vine`.

Make sure you're checking out the latest main branch! I pushed a lot of new commits these days. 💗

**Reply by CodeGetters** [reply]
yes,you're right. 😉The dist directory is not found in the packages/vite-plugin directory, apparently because it is not included when the package command is not used. Can you tell me how to package this directory separately, since the related command is not found in package.json🤔

**Reply by ShenQingchuan** [reply]
> yes,you're right. 😉The dist directory is not found in the packages/vite-plugin directory, apparently because it is not included when the package command is not used. Can you tell me how to package this directory separately, since the related command is not found in package.json🤔


As our expectation, after you run `pnpm run dev`, we'll compile the following 3 packages:

1. `compiler`
2. `vite-plugn`
3. `vue-vine`

When the output in terminal stopped, there's supposed to be a `dist` folders in each package.

What's your output after running `pnpm run dev`?

**Reply by ShenQingchuan** [reply]
@CodeGetters 

Apologies for my mistake, one of the steps in the "try the demo" guide in the previous Readme was missed:

After `pnpm install` you need to `pnpm run build` first. 

Please try again by pulling the latest main branch.

**Reply by CodeGetters** [reply]
@ShenQingchuan 
so cool😘 Everything seems normal now✔ thank you!
